tests/unit/test_mxnet.py

```diff
@@ -100,17 +102,54 @@ def skip_if_not_mms_version(mxnet_version):
         pytest.skip("Skipping because this version does not use MMS")
 
 
-def _get_full_image_uri(version, repo=IMAGE_REPO_NAME, processor="cpu", py_version="py2"):
-    return FULL_IMAGE_URI.format(repo, version, processor, py_version)
+def _use_dlc(version, py_version, with_ei=False):
+    # TODO: in refactor of image uri, make less brittle. currently matches min version map
+    version_list = [int(s) for s in version.split(".")]
+    if py_version == "py3" or with_ei:
+        return version_list >= [1, 4, 1][: len(version_list)]
+    if py_version == "py2":
+        return version_list >= [1, 6, 0][: len(version_list)]
+    return False
 
 
-def _get_full_image_uri_with_ei(version, repo=IMAGE_REPO_NAME, processor="cpu", py_version="py2"):
-    return FULL_IMAGE_URI.format("{}-eia".format(repo), version, processor, py_version)
+def _get_account(version, py_version, with_ei=False):
+    return ACCOUNT_DLC if _use_dlc(version, py_version, with_ei) else ACCOUNT
 
 
-def _create_train_job(version):
+def _get_image_repo_name(version, py_version):
+    return IMAGE_REPO_NAME_DLC if _use_dlc(version, py_version) else IMAGE_REPO_NAME
+
+
+def _get_image_repo_serving_name(version, py_version, with_ei=False):
+    use_dlc = _use_dlc(version, py_version, with_ei)
+    return IMAGE_REPO_SERVING_NAME_DLC if use_dlc else IMAGE_REPO_SERVING_NAME
+
+
+def _get_full_image_uri(version, py_version, proc="cpu", repo=None, with_ei=False):
+    account = _get_account(version, py_version, with_ei)
+    if repo is None:
+        repo = _get_image_repo_name(version, py_version)
+    ei = "-eia" if with_ei else ""
+
+    full_uri = "{account}.dkr.ecr.us-west-2.amazonaws.com/{repo}{ei}:{version}-{proc}-{py_version}"
+    return full_uri.format(
+        account=account, repo=repo, ei=ei, version=version, proc=proc, py_version=py_version
+    )
+
+
+def _get_full_image_uri_with_ei(version, py_version, proc="cpu", repo=None):
+    return _get_full_image_uri(version, py_version, proc=proc, repo=repo, with_ei=True)
```

Comment:
this stuff should already be tested in https://github.com/aws/sagemaker-python-sdk/blob/master/tests/unit/test_fw_utils.py. (though that file could certainly use its own overhaul...)

rather than duplicating the create_image_uri stuff here, just mock it instead (and make it return something boring that's easy to match against), and check that the function is called with the right arguments. we shouldn't duplicate the testing of the image uri generation logic in this file.

example: https://github.com/aws/sagemaker-python-sdk/blob/zwei/tests/unit/sagemaker/tensorflow/test_tfs.py#L73

Reply:
agreed. will clean this up